### PR TITLE
[Spritelab][Poem Bot HOC] Initial Poem Bank prototype

### PR DESCRIPTION
**Pull request by ajpal**
Behind experiment flag `poemBot`

Modal to pick a poem from the poem bank
image

Selected poem is displayed under the playspace:
image

For now poem bank is defined in `poems.json` but my guess is that at some point we may want to build a LB edit page similar to http://levelbuilder-studio.code.org/datasets for managing poems.

The UI is completely non-finalized, the main purpose is to just get a working prototype for Amy BW to be able to write lessons around. 

## Links
[jira](https://codedotorg.atlassian.net/browse/STAR-1593)

## Testing story
Going to hold off on unit tests until we finalize design

## Follow-up work
- Make poem bank per-level configurable

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/p5lab/P5LabVisualizationColumn.jsx [hunk 1]
```diff
@@ -57,6 +59,9 @@ class P5LabVisualizationColumn extends React.Component {
 
   constructor(props) {
     super(props);
+    this.spritelabPoemBotExperiment = experiments.isEnabled(
```
Comment (maddiedierker) [inline apps/src/p5lab/P5LabVisualizationColumn.jsx:62]: not sure how much we care about this, but we could check `isSpritelab` as part of this conditional to make sure `<PoemBank/>` is only displayed in spritelab

File: apps/src/p5lab/spritelab/PoemBank.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+class PoemTile extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    author: PropTypes.string.isRequired,
+    lines: PropTypes.arrayOf(PropTypes.string).isRequired,
+    onSelect: PropTypes.func.isRequired
+  };
+
+  selectPoem() {
+    this.props.onSelect(this.props.name, this.props.author, this.props.lines);
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/PoemBank.jsx]: nit: this line would be more readable if we first destructured onSelect, name, author, and lines from props

File: apps/src/p5lab/spritelab/PoemBank.jsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import PropTypes from 'prop-types';
+import BaseDialog from '@cdo/apps/templates/BaseDialog.jsx';
+import poems from './poems.json';
+
+export const INITIAL_STATE = {
+  isOpen: false,
+  selectedPoem: null
+};
+
+class PoemTile extends React.Component {
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/PoemBank.jsx]: this component doesn't have any state, so i think we should refactor it to a functional component. react is standardizing on functional over class components, so i'm trying to move us toward functional components wherever we can

Comment (maddiedierker) [reply apps/src/p5lab/spritelab/PoemBank.jsx]: for functional components, i usually set them up like this:


```javascript
function MyComponent(props) {
  function otherFunction() {
    // i still have access to `props`
  }

  return <div>{props.text}</div>;
}

MyComponent.propTypes = {
  text: PropTypes.string.isRequired
};

MyComponent.defaultProps = {
  text: 'Hello, world!'
};
```

File: apps/src/p5lab/spritelab/PoemBank.jsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        {this.props.lines.map((line, i) => (
+          <div key={line + i}>{line}</div>
+        ))}
+      </div>
+    );
+  }
+}
+
+export default class PoemBank extends React.Component {
+  static propTypes = {};
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/PoemBank.jsx]: nit: this line can be removed if we don't have any props for this component

File: apps/src/p5lab/spritelab/PoemBank.jsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    display: 'inline-block',
+    width: '95%'
+  },
+  modalBody: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%'
+  },
+  poemTile: {
+    border: '1px solid black',
```
Comment (maddiedierker) [inline apps/src/p5lab/spritelab/PoemBank.jsx:100]: nit: when we finalize this, we should use colors from our color constants

Comment (ajpal) [reply apps/src/p5lab/spritelab/PoemBank.jsx:100]: Yeah definitely, this is not at all the finalized design- I just added a black border so I could see the components better :)

File: apps/src/p5lab/spritelab/PoemBank.jsx (hunk 3 above)
Comment (ajpal) [reply apps/src/p5lab/spritelab/PoemBank.jsx]: thanks! that feels very clean for a component as small and simple as this :)

**Review by maddiedierker — APPROVED** [review]

**Pull request merged**